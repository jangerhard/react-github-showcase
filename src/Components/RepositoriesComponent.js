import PropTypes from 'prop-types';
import React from 'react';
import RepositoryComponent from "./RepositoryComponent";

const RepositoriesComponent = ({ repos }) => (
    <div className="repos" style={{}}>
        {repos.map(function (repo) {

            const commits = repo.node.ref !== null ? repo.node.ref.target.history.nodes : [];

            return <RepositoryComponent
                key={repo.node.url}
                name={repo.node.name}
                commits={commits}
                url={repo.node.url} />;
        })}
    </div>
);

RepositoriesComponent.propTypes = {
    repos: PropTypes.array.isRequired
};

export default RepositoriesComponent;