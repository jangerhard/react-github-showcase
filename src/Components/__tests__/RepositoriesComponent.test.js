import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import RepositoriesComponent from "../RepositoriesComponent";

const repos = [
    {
        "node": {
            "name": "react-github-showcase",
            "url": "https://github.com/jangerhard/react-github-showcase",
            "ref": {
                "target": {
                    "history": {
                        "nodes": [
                            {
                                "author": {
                                    "name": "Jan Gerhard Schoepp",
                                    "avatarUrl": "https://avatars2.githubusercontent.com/u/6482205?v=4"
                                },
                                "message": "Update README.md",
                                "abbreviatedOid": "3ce6669",
                                "committedDate": "2018-05-04T19:59:54Z",
                                "url": "https://github.com/jangerhard/react-github-showcase/commit/3ce6669da1f699e254a4aaccd5964863915547a0"
                            },
                            {
                                "author": {
                                    "name": "jangerhard",
                                    "avatarUrl": "https://avatars2.githubusercontent.com/u/6482205?v=4"
                                },
                                "message": "0.0.14",
                                "abbreviatedOid": "0106258",
                                "committedDate": "2018-05-04T17:40:37Z",
                                "url": "https://github.com/jangerhard/react-github-showcase/commit/01062589e243a9044b04c49f86775937af273fe8"
                            }
                        ]
                    }
                }
            }
        }
    },
    {
        "node": {
            "name": "PersonalGatsbyWebsite",
            "url": "https://github.com/jangerhard/PersonalGatsbyWebsite",
            "ref": {
                "target": {
                    "history": {
                        "nodes": [
                            {
                                "author": {
                                    "name": "Jan Gerhard Schoepp",
                                    "avatarUrl": "https://avatars2.githubusercontent.com/u/6482205?v=4"
                                },
                                "message": "Merge pull request #2 from jangerhard/cleanup_components\n\nCleanup components",
                                "abbreviatedOid": "66aea85",
                                "committedDate": "2018-05-03T18:43:21Z",
                                "url": "https://github.com/jangerhard/PersonalGatsbyWebsite/commit/66aea85be45f73383509466e8d48e376da3cb088"
                            },
                            {
                                "author": {
                                    "name": "jangerhard",
                                    "avatarUrl": "https://avatars2.githubusercontent.com/u/6482205?v=4"
                                },
                                "message": "Use shorthand for components",
                                "abbreviatedOid": "63c5423",
                                "committedDate": "2018-05-03T18:38:07Z",
                                "url": "https://github.com/jangerhard/PersonalGatsbyWebsite/commit/63c5423e955f0b5064d2abe556d75d568d9f849a"
                            }
                        ]
                    }
                }
            }
        }
    }
];

const reposWithNoCommits = [
    {
        "node": {
            "name": "react-github-showcase",
            "url": "https://github.com/jangerhard/react-github-showcase",
            "ref": null
        }
    },
    {
        "node": {
            "name": "PersonalGatsbyWebsite",
            "url": "https://github.com/jangerhard/PersonalGatsbyWebsite",
            "ref": null
        }
    }
];

test('renders repositories correctly when loading', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <RepositoriesComponent
            repos={repos}
            loading
        />
    );
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
});
test('renders repositories correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <RepositoriesComponent
            repos={repos}
        />
    );
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
});

test('renders repositories with no commits ', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <RepositoriesComponent
            repos={reposWithNoCommits}
        />
    );
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
});