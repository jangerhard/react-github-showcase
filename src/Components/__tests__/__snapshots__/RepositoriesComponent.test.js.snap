// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders repositories correctly 1`] = `
<ReactPlaceholder
  color="#CDCDCD"
  customPlaceholder={
    <div
      className="repos"
      style={
        Object {
          "width": "30em",
        }
      }
    >
      <TextBlock
        color="grey"
        rows={5}
        widths={
          Array [
            97,
            100,
            94,
            90,
            98,
            95,
            98,
            40,
          ]
        }
      />
    </div>
  }
  delay={0}
  ready={true}
  type="text"
>
  <div
    className="repos"
    style={Object {}}
  >
    <RepositoryComponent
      commits={
        Array [
          Object {
            "abbreviatedOid": "3ce6669",
            "author": Object {
              "avatarUrl": "https://avatars2.githubusercontent.com/u/6482205?v=4",
              "name": "Jan Gerhard Schoepp",
            },
            "committedDate": "2018-05-04T19:59:54Z",
            "message": "Update README.md",
            "url": "https://github.com/jangerhard/react-github-showcase/commit/3ce6669da1f699e254a4aaccd5964863915547a0",
          },
          Object {
            "abbreviatedOid": "0106258",
            "author": Object {
              "avatarUrl": "https://avatars2.githubusercontent.com/u/6482205?v=4",
              "name": "jangerhard",
            },
            "committedDate": "2018-05-04T17:40:37Z",
            "message": "0.0.14",
            "url": "https://github.com/jangerhard/react-github-showcase/commit/01062589e243a9044b04c49f86775937af273fe8",
          },
        ]
      }
      name="react-github-showcase"
      url="https://github.com/jangerhard/react-github-showcase"
    />
    <RepositoryComponent
      commits={
        Array [
          Object {
            "abbreviatedOid": "66aea85",
            "author": Object {
              "avatarUrl": "https://avatars2.githubusercontent.com/u/6482205?v=4",
              "name": "Jan Gerhard Schoepp",
            },
            "committedDate": "2018-05-03T18:43:21Z",
            "message": "Merge pull request #2 from jangerhard/cleanup_components

Cleanup components",
            "url": "https://github.com/jangerhard/PersonalGatsbyWebsite/commit/66aea85be45f73383509466e8d48e376da3cb088",
          },
          Object {
            "abbreviatedOid": "63c5423",
            "author": Object {
              "avatarUrl": "https://avatars2.githubusercontent.com/u/6482205?v=4",
              "name": "jangerhard",
            },
            "committedDate": "2018-05-03T18:38:07Z",
            "message": "Use shorthand for components",
            "url": "https://github.com/jangerhard/PersonalGatsbyWebsite/commit/63c5423e955f0b5064d2abe556d75d568d9f849a",
          },
        ]
      }
      name="PersonalGatsbyWebsite"
      url="https://github.com/jangerhard/PersonalGatsbyWebsite"
    />
  </div>
</ReactPlaceholder>
`;

exports[`renders repositories correctly when loading 1`] = `
<ReactPlaceholder
  color="#CDCDCD"
  customPlaceholder={
    <div
      className="repos"
      style={
        Object {
          "width": "30em",
        }
      }
    >
      <TextBlock
        color="grey"
        rows={5}
        widths={
          Array [
            97,
            100,
            94,
            90,
            98,
            95,
            98,
            40,
          ]
        }
      />
    </div>
  }
  delay={0}
  ready={false}
  type="text"
>
  <div
    className="repos"
    style={Object {}}
  >
    <RepositoryComponent
      commits={
        Array [
          Object {
            "abbreviatedOid": "3ce6669",
            "author": Object {
              "avatarUrl": "https://avatars2.githubusercontent.com/u/6482205?v=4",
              "name": "Jan Gerhard Schoepp",
            },
            "committedDate": "2018-05-04T19:59:54Z",
            "message": "Update README.md",
            "url": "https://github.com/jangerhard/react-github-showcase/commit/3ce6669da1f699e254a4aaccd5964863915547a0",
          },
          Object {
            "abbreviatedOid": "0106258",
            "author": Object {
              "avatarUrl": "https://avatars2.githubusercontent.com/u/6482205?v=4",
              "name": "jangerhard",
            },
            "committedDate": "2018-05-04T17:40:37Z",
            "message": "0.0.14",
            "url": "https://github.com/jangerhard/react-github-showcase/commit/01062589e243a9044b04c49f86775937af273fe8",
          },
        ]
      }
      name="react-github-showcase"
      url="https://github.com/jangerhard/react-github-showcase"
    />
    <RepositoryComponent
      commits={
        Array [
          Object {
            "abbreviatedOid": "66aea85",
            "author": Object {
              "avatarUrl": "https://avatars2.githubusercontent.com/u/6482205?v=4",
              "name": "Jan Gerhard Schoepp",
            },
            "committedDate": "2018-05-03T18:43:21Z",
            "message": "Merge pull request #2 from jangerhard/cleanup_components

Cleanup components",
            "url": "https://github.com/jangerhard/PersonalGatsbyWebsite/commit/66aea85be45f73383509466e8d48e376da3cb088",
          },
          Object {
            "abbreviatedOid": "63c5423",
            "author": Object {
              "avatarUrl": "https://avatars2.githubusercontent.com/u/6482205?v=4",
              "name": "jangerhard",
            },
            "committedDate": "2018-05-03T18:38:07Z",
            "message": "Use shorthand for components",
            "url": "https://github.com/jangerhard/PersonalGatsbyWebsite/commit/63c5423e955f0b5064d2abe556d75d568d9f849a",
          },
        ]
      }
      name="PersonalGatsbyWebsite"
      url="https://github.com/jangerhard/PersonalGatsbyWebsite"
    />
  </div>
</ReactPlaceholder>
`;

exports[`renders repositories with no commits  1`] = `
<ReactPlaceholder
  color="#CDCDCD"
  customPlaceholder={
    <div
      className="repos"
      style={
        Object {
          "width": "30em",
        }
      }
    >
      <TextBlock
        color="grey"
        rows={5}
        widths={
          Array [
            97,
            100,
            94,
            90,
            98,
            95,
            98,
            40,
          ]
        }
      />
    </div>
  }
  delay={0}
  ready={true}
  type="text"
>
  <div
    className="repos"
    style={Object {}}
  >
    <RepositoryComponent
      commits={Array []}
      name="react-github-showcase"
      url="https://github.com/jangerhard/react-github-showcase"
    />
    <RepositoryComponent
      commits={Array []}
      name="PersonalGatsbyWebsite"
      url="https://github.com/jangerhard/PersonalGatsbyWebsite"
    />
  </div>
</ReactPlaceholder>
`;
